nautobot/core/models/querysets.py

```diff
@@ -125,16 +125,11 @@ def restrict(self, user, action="view"):
         # Filter the queryset to include only objects with allowed attributes
         else:
             attrs = Q()
-            for perm_attrs in user._object_perm_cache[permission_required]:
-                if isinstance(perm_attrs, list):
-                    for p in perm_attrs:
-                        attrs |= Q(**p)
```

Comment:
Does this case need to be handled specially by `qs_filter_from_constraints`? This would I believe be for a case where we have an objectpermission with constraints like `[{"user": "$user"}, {"job__name": "MyJob"}]`, i.e OR conditions in the constraints.

Reading the code carefully, this case **might** be dead code, as `user._object_perm_cache` is a list constructed by calling `ObjectPermission.list_constraints()` and appending the result, so it **might** be already exploding such constraints out into multiple separate entries in `_object_perm_cache` - but please test and confirm.

Reply:
Regarding this, if we have a constraint like `{"user":"$user"}` and not use the `qs_filter_from_constraints` I get a `ValidationError at /extras/object-changes/` with messsage `['“$user” is not a valid UUID.']`. 
If I don't have a constraint like the above, the original code was working as expected.

Reply:
I think maybe we're talking past one another :smile:. I definitely get that `qs_filter_from_constraints` is necessary to handle any `$user` tokens in the constraints. My concern was whether the way it's being used here also correctly handles the "list of constraints in a single ObjectPermission" case that I mentioned in my comment above. As I said, I **think** it should work, but wanted to make sure we test that case as well.

Reply:
Alright, tested it with a list of constraints like `[{"user": "$user"}, {"action": "delete"}]` and I also added one test case for it. 
I noticed that with the the example I just mentioned, the user sees his all of his own logs, and logs of other users only if the action is `delete`.

Reply:
Sounds correct to me. Thanks!